<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Use quanTIseq to deconvolute a gene expression matrix. — deconvolute_quantiseq.default • immunedeconv</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Use quanTIseq to deconvolute a gene expression matrix. — deconvolute_quantiseq.default" />

<meta property="og:description" content="Source code from https://github.com/FFinotello/quanTIseq" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">immunedeconv</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/immunedeconv.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/grst/immune_deconvolution_methods">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Use quanTIseq to deconvolute a gene expression matrix.</h1>
    <small class="dont-index">Source: <a href='https://github.com/grst/immune_deconvolution_methods/blob/master/R/quantiseq.R'><code>R/quantiseq.R</code></a></small>
    <div class="hidden name"><code>deconvolute_quantiseq.default.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Source code from https://github.com/FFinotello/quanTIseq</p>
    
    </div>

    <pre class="usage"><span class='fu'>deconvolute_quantiseq.default</span>(<span class='no'>mix.mat</span>, <span class='kw'>arrays</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>signame</span> <span class='kw'>=</span> <span class='st'>"TIL10"</span>,
  <span class='kw'>tumor</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>mRNAscale</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"lsei"</span>, <span class='kw'>btotalcells</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>rmgenes</span> <span class='kw'>=</span> <span class='st'>"unassigned"</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>mix.mat</th>
      <td><p>table with the gene TPM (or microarray expression values) for all samples to be deconvoluted
(Gene symbols on the first column and sample IDs on the first row). Expression data must be on non-log scale</p></td>
    </tr>
    <tr>
      <th>arrays</th>
      <td><p>specifies whether expression data are from microarrays (instead of RNA-seq).
If TRUE, the "--rmgenes" parameter is set to "none".</p></td>
    </tr>
    <tr>
      <th>signame</th>
      <td><p>name of the signature matrix. Currently only <code>TIL10</code> is available.</p></td>
    </tr>
    <tr>
      <th>tumor</th>
      <td><p>specifies whether expression data are from tumor samples. If TRUE, signature genes
with high expression in tumor samples are removed (see 1).
Default: FALSE.</p></td>
    </tr>
    <tr>
      <th>mRNAscale</th>
      <td><p>specifies whether cell fractions must be scaled to account for cell-type-specific
mRNA content.
Default: TRUE.</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>deconvolution method to be used: "hampel", "huber", or "bisquare" for robust regression
with Huber, Hampel, or Tukey bisquare estimators, respectively, or "lsei" for constrained
least squares regression. The fraction of uncharacterized cells ("other") is computed only
by the "lsei" method.
Default: "lsei".</p></td>
    </tr>
    <tr>
      <th>btotalcells</th>
      <td><p>compute cell densities instead of fractions
Default: FALSE</p></td>
    </tr>
    <tr>
      <th>rmgenes</th>
      <td><p>Default: "default" for RNAseq, "none" for microArray data</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>F. Finotello, C. Mayer, C. Plattner, G. Laschober, D. Rieder,
H. Hackl, A. Krogsdam, W. Posch, D. Wilflingseder, S. Sopper, M. Jsselsteijn,
D. Johnsons, Y. Xu, Y. Wang, M. E. Sanders, M. V. Estrada, P. Ericsson-Gonzalez,
J. Balko, N. F. de Miranda, Z. Trajanoski. "quanTIseq: quantifying immune contexture of human tumors".
bioRxiv 223180. https://doi.org/10.1101/223180.</p>
    

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#details">Details</a></li>
          </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Gregor Sturm.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

