<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ immunedeconv</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
<meta property="og:title" content="Function reference" />
<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">immunedeconv</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/getting_started.html">R Notebook</a>
    </li>
    <li>
      <a href="../articles/immune_deconvolution_methods.html">Collection of Methods for immune cell deconvolution from bulk RNAseq</a>
    </li>
    <li>
      <a href="../articles/single_cell_dataset_schelker.html">Preparing the single cell dataset by Schelker et al. </a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/grst/immune_deconvolution_methods">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9">
    <div class="page-header">
      <h1>
        Reference
        <small>version&nbsp;0.1</small>
      </h1>
    </div>

    <div class="contents">
      <table class="ref-index">

      <colgroup>
        <col class="alias" />
        <col class="title" />
      </colgroup>

      <tbody>
        <tr>
          <th colspan="2">
            <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
            <p class="section-desc"></p>
          </th>
        </tr>
        <tr>
          <!--  -->
          <td>
            <p><code><a href="GetFractions.Abbas.html">GetFractions.Abbas()</a></code> </p>
          </td>
          <td><p>Constrained regression method implemented in Abbas et al., 2009</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="annotate_cell_type.html">annotate_cell_type()</a></code> </p>
          </td>
          <td><p>Annotate unified cell_type names</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="available_datasets.html">available_datasets</a></code> </p>
          </td>
          <td><p>Available methods and datasets</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="cell_type_mapping.html">cell_type_mapping</a></code> </p>
          </td>
          <td><p>Table mapping the cell types from methods/datasets to a single, controlled vocabulary</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="cell_type_tree.html">cell_type_tree</a></code> </p>
          </td>
          <td><p>Available cell types in the controlled vocabulary organized as a lineage tree.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="check_cancer_types.html">check_cancer_types()</a></code> </p>
          </td>
          <td><p>process batch table and check cancer types.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="deconvolute.html">deconvolute.default()</a></code> <code><a href="deconvolute.html">deconvolute()</a></code> </p>
          </td>
          <td><p>Perform an immune cell deconvolution on a dataset.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="deconvolute_quantiseq.default.html">deconvolute_quantiseq.default()</a></code> </p>
          </td>
          <td><p>Use quanTIseq to deconvolute a gene expression matrix.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="deconvolute_timer.html">deconvolute_timer()</a></code> </p>
          </td>
          <td><p>Deconvolute using the TIMER technique</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="deconvolution_methods.html">deconvolution_methods</a></code> </p>
          </td>
          <td><p>list of supported immune deconvolution methods</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="eset_to_matrix.html">eset_to_matrix()</a></code> </p>
          </td>
          <td><p>convert an <code>Biobase::ExpressionSet</code> to a gene-expression matrix.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="example_gene_expression_matrix.html">example_gene_expression_matrix</a></code> </p>
          </td>
          <td><p>example gene expression dataset</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="find_children.html">find_children()</a></code> </p>
          </td>
          <td><p>recursively traverse the cell type tree to resolve the mapping.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="immune_deconvolution_methods.html">immune_deconvolution_methods</a></code> </p>
          </td>
          <td><p>Collection of immune cell deconvolution methods.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="make_bulk_eset.html">make_bulk_eset()</a></code> </p>
          </td>
          <td><p>make a random expression set from a single-cell dataset</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="make_random_bulk.html">make_random_bulk()</a></code> </p>
          </td>
          <td><p>make a random bulk sample from a single-cell dataset</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="map_cell_types.html">map_cell_types()</a></code> </p>
          </td>
          <td><p>Use a tree-hierarchy to map cell types among different methods.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="map_result_to_celltypes.html">map_result_to_celltypes()</a></code> </p>
          </td>
          <td><p>Map a result table as generated by <code>deconvolute</code> to a user-defined list of cell types.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="set_cibersort_binary.html">set_cibersort_binary()</a></code> </p>
          </td>
          <td><p>set Path to CIBERSORT R script</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="set_cibersort_mat.html">set_cibersort_mat()</a></code> </p>
          </td>
          <td><p>set Path to CIBERSORT matrix file</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="timer_available_cancers.html">timer_available_cancers</a></code> </p>
          </td>
          <td><p>TIMER signatures are cancer specific. This is the list of available cancer types.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="xCell.data.html">xCell.data</a></code> </p>
          </td>
          <td><p>Data object from xCell.</p></td>
        </tr>
      </tbody>
      </table>
    </div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#section-all-functions">All functions</a></li>
    </ul>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Gregor Sturm.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>

